src/naive_bayes/mod.rs

```diff
@@ -0,0 +1,239 @@
+use crate::error::Failed;
+use crate::linalg::BaseVector;
+use crate::linalg::Matrix;
+use crate::math::num::RealNumber;
+use std::marker::PhantomData;
+
+/// Distribution used in the Naive Bayes classifier.
+pub trait NBDistribution<T: RealNumber, M: Matrix<T>> {
```

Comment:
Let's make this trait pub(crate). This trait and its implementations are not useful outside of NB and  I think we can simplify API with another class that is similar to Scikit learn's CategoricalNB. This new class should combine distribution with `BaseNaiveBayes ` to make sure we can do
```
let nb = BaseNaiveBayes::fit(&x, &y, Default::default()).unwrap();
```
instead of
```
let distribution = CategoricalNB::<f64>::fit(&x, &y, alpha).unwrap();
let nb = BaseNaiveBayes::fit(distribution).unwrap();
```

Reply:
Sorry for confusion. First code snippet should be:
```
let nb = CategoricalNB::fit(&x, &y, Default::default()).unwrap();
```